Makefile

```diff
@@ -36,6 +36,11 @@ SRC_CFLAGS := -pedantic
 TEST_CFLAGS := -Wno-unused-parameter -iquote$(SRC_DIR)
 SP := strip
 
+# GCC fails with decode.h zero-length format strings
+# GCC fails with dis.c notimplemented instructions having unused parameters
+# GCC fails with dis.c notimplemented instructions having unused variables
+CFLAGS += -Wno-format-zero-length -Wno-unused-parameter -Wno-unused-variable
```

Comment:
early on I used to regularly check against gcc on Linux (Ubuntu specifically) but haven't reinstalled docker since upgrading my Mac last year so I'm not surprised warnings have snuck in. currently I differentiate based on `uname`  and I just assume if Darwin then we're using clang, otherwise assume gcc. That's probably not the best way to do it.

I would drop the unused-param and unused-variable flags since those warnings are temporary and if there's a permanent situation where I have an unused parameter I will `(void)` it out explicitly. the current warnings are there specifically to get my IDE to draw attention to it so I don't forget about them (since this is a personal project I don't bother sticking WIP code in a separate branch so fair warning `main` isn't always in a clean, releasable state).

I didn't realize format-zero-length was an issue though, that's a good one to add. would it be worth making it specific to gcc or just include it for everything? clearly clang isn't complaining but that also means it doesn't hurt